src/runtime/components/Divider.vue

```diff
@@ -0,0 +1,69 @@
+<script lang="ts">
+import { tv, type VariantProps } from 'tailwind-variants'
+import type { SeparatorProps } from 'radix-vue'
+import type { AvatarProps } from './Avatar.vue'
+import type { AppConfig } from '@nuxt/schema'
+import _appConfig from '#build/app.config'
+import theme from '#build/ui/divider'
+
+const appConfig = _appConfig as AppConfig & { ui: { divider: Partial<typeof theme> } }
+
+const divider = tv({ extend: tv(theme), ...(appConfig.ui?.divider || {}) })
+
+type DividerVariants = VariantProps<typeof divider>
+
+export interface DividerProps extends Omit<SeparatorProps, 'asChild'> {
+  avatar?: AvatarProps
+  class?: any
+  color?: DividerVariants['color']
+  decorative?: boolean
+  icon?: string
+  label?: string
+  orientation?: DividerVariants['orientation']
+  size?: DividerVariants['size']
+  type?: DividerVariants['type']
+  ui?: Partial<typeof divider.slots>
+}
+
+export interface DividerSlots {
+  default(): any
+}
+</script>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { Separator } from 'radix-vue'
+import { UAvatar, UIcon } from '#components'
+
+const props = withDefaults(defineProps<DividerProps>(), {
+  as: 'div',
+  orientation: 'horizontal',
+  size: '2xs'
```

Comment:
The default `size` and `orientation` should be in the `defaultVariants` of the config.